### PR TITLE
Reworked and added threatened dreams quest

### LINKED ISSUE #614: [MAP] Wrong item and tiles on map - Threatened Dreams Quest
### Priority

Medium

### Area

- [X] Datapack
- [ ] Source
- [X] Map
- [ ] Other

### What happened?

The request in a certain part request you to freedom some fairies, but one of the fairies is not on local, instead there is **a test wand of voodoo.**

**Position: 33621, 32214, 8**
image

The correct item ID of the Fairy is:
Client ID: **25796**

Another part of the quest requires the player to swimming to an isle, but some of the IDs around the isle doesn't allow swimming:
**Position: 33547, 32250, 7**
image

**The correct ID should be 632**

In this location **Position: 33496, 32196, 7**
There is a ferns, this item must be removed, the location is actually to be clean, it will have an event that will create a raven bush during the night and remove on sunrise.
image

### What OS are you seeing the problem on?

Ubuntu 20.04

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #628: [MAP] A border item is covering a quest item
### Priority

Medium

### Area

- [X] Datapack
- [ ] Source
- [X] Map
- [ ] Other

### What happened?

During the workflow of the Threatened Dream quest, there is a part where you need to grab a herb, this herb only shows up during the night.

But while the herb is being created a border on the map is still covering the herb, making it impossible to grab.
A simple solution would be by just removing this border (Forest Floor) from the map. I have done it by command /r and I was able to move forward with the quest.

**Position: 33496, 32196, 7**
image

### What OS are you seeing the problem on?

Ubuntu 20.04

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sircurse**
Rework of the initial quest that give access to Feyrist area and added additional quests to the story line.

## Type of change

Please delete options that are not relevant.

  - [X] Bug fix (non-breaking change which fixes an issue)
  - [X] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] This change requires a documentation update

## How Has This Been Tested

The scripts was tested by me and players from my server.

## Checklist

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my own code
  - [X] I checked the PR checks reports
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: data/lib/core/storages.lua [hunk 1]
```diff
@@ -2235,7 +2208,61 @@ Storage = {
...
+				Feathers4 = 50278,
+				Feathers5 = 50279
+				},
+			Mission02 = {50280, -- Nightmare Intruders
+				FrazzlemawsCount = 50281,
+				EnfeebledCount = 50282,
+				KroazurAccess = 50283,
+				KroazurTimer = 50284,
+				KroazurKill = 50285,
+				DarkMoomMirror = 50286,
```
Comment (guispiller) [inline data/lib/core/storages.lua]: fix Typo "moon" here and in other scripts

File: data/lib/core/storages.lua [hunk 2]
```diff
@@ -2235,7 +2208,61 @@ Storage = {
...
+				MoonMirrorPos02 = 50295,
+				MoonMirrorPos03 = 50296,
+				MoonMirrorPos04 = 50297,
+				MoonMirrorPos05 = 50298,
+				ChargedSunCacther = 50299,
+				SunCactherPos01 = 50300,
+				SunCactherPos02 = 50301,
+				SunCactherPos03 = 50302,
+				SunCactherPos04 = 50303,
+				SunCactherPos05 = 50304,
```
Comment (guispiller) [inline data/lib/core/storages.lua]: fix typo "catcher" in this 6 storages above and in other scripts

Comment (guispiller) [inline data/lib/core/storages.lua]: fix typo "catcher" in this 6 storages above

File: data/startup/tables/tile.lua [hunk 3]
```diff
@@ -227,6 +227,61 @@ TileAction = {
...
+	[25029] = {
+		itemId = 10840,
+		itemPos = {
+			{x = 33619, y = 32304, z = 9},
+			{x = 33591, y = 32321, z = 10}
+		}
+	},
 	-- Others actions
 	-- Threatened Dreams Quest
 	[50301] = {
```
Comment (guispiller) [inline data/startup/tables/tile.lua]: these AIDs from 50301 to 50305  should be removed, they are conflicting with the new ones

Comment (guispiller) [inline data/startup/tables/tile.lua:287]: this AIDs should be removed, they are conflicting with the new ones

**Review by guispiller — CHANGES_REQUESTED** [review]
Small typo fixes

There's also a script that's conflicting with another you created and should be deleted: data/scripts/movements/quests/feyrist/kroazur_entrance.lua

**Review by majestyotbr — APPROVED** [review]

File: data/lib/core/storages.lua [hunk 4]
```diff
@@ -2344,8 +2317,62 @@ Storage = {
...
+					KroazurTimer = 75765,
+					KroazurKill = 75766,
+					DarkMoonMirror = 75767,
+					FairiesCounter = 75768,
+					Fairy01 = 75769,
+					Fairy02 = 75770,
+					Fairy03 = 75771,
+					Fairy04 = 75772,
+					Fairy05 = 75773,
+					ChargedMoomMirror = 75774,
```
Comment (guispiller) [inline data/lib/core/storages.lua]: There's just this storage left to fix typos, the rest is fine. Fix typo "moom" here, and on scripts that have this same storage.

**Review by guispiller — CHANGES_REQUESTED** [review]

**Review by guispiller — APPROVED** [review]

**Review by majestyotbr — APPROVED** [review]

**Comment by majestyotbr** [comment]
Thanks for your contribution!

**Comment by majestyotbr** [comment]
Updated otservbr-global map available in:
https://majestyotbr.gitbook.io/opentibiabr/projects/otservbr-global#updated-otservbr-global-map

**Pull request merged**